src/Sylius/Bundle/ApiBundle/Validator/Constraints/AddingEligibleProductVariantToCartValidator.php

```diff
@@ -85,7 +86,31 @@ public function validate($value, Constraint $constraint)
             return;
         }
 
-        if (!$this->availabilityChecker->isStockSufficient($productVariant, $value->quantity)) {
+        /** @var OrderInterface|null $cart */
+        $cart = $this->orderRepository->findCartByTokenValue($value->getOrderTokenValue());
+        Assert::notNull($cart);
+
+        $units = $cart->getItemUnits();
+
+        $sameOrderItemsCounter = 0;
+
+        if ($productVariant->isTracked() && !$units->isEmpty()) {
+            foreach ($units as $unit) {
+                /** @var OrderItemInterface $orderItem */
+                $orderItem = $unit->getOrderItem();
+
+                /** @var ProductVariantInterface */
+                $variant = $orderItem->getVariant();
+
+                $code = $variant->getCode();
+
+                if ($value->productVariantCode = $code) {
+                    ++$sameOrderItemsCounter;
+                }
+            }
```

Comment:
I am wondering about the performance of this solution. In cart, we can have a really big amount of item units, and every addition to the cart will iterate over the entire cart...